.github/workflows/test.yml

```diff
@@ -26,9 +26,36 @@ jobs:
         script: |
             core.setFailed('Tagged commit does not match main')
 
-  test:
+  version-bump:
     runs-on: ubuntu-latest
     needs: [validate]
+    env:
+      VERSION: $GITHUB_REF_NAME
+    steps:
+      - name: Checkout main branch
+        uses: actions/checkout@v2
+        with:
+          ref: 'main'
+          token: ${{ secrets.CONTENTS_RW }}
+      - name: Set version for commit message
+        id: bumped_version
+        run: echo "RELEASE_VERSION=$GITHUB_REF_NAME" >> $GITHUB_OUTPUT
+      - name: Update version
+        run: python scripts/update_version.py ${{ env.VERSION }}
+      - name: Auto-commit changes
+        uses: stefanzweifel/git-auto-commit-action@3ea6ae190baf489ba007f7c92608f33ce20ef04a #v4.16.0
+        with:
+          commit_message: Bump version to ${{ steps.bumped_version.outputs.RELEASE_VERSION }} [skip ci]
+      - name: Merge changes to main
+        env:
+          key: ${{ secrets.ADMIN_RW }}
+        run: |
+          git tag -f ${{ env.VERSION }}
+          git push -f origin ${{ env.VERSION }}
```

Comment:
This didn't work?

```suggestion
          git tag -f "$GITHUB_REF_NAME"
          git push -f origin "$GITHUB_REF_NAME"
```

Reply:
Ah good call, the commit message kept throwing me off: `Bump version to "$GITHUB_REF_NAME" [skip ci]`. Thank you!